File: Euclidean_algorithm/Javascript/jcla1/euclidean.js
function gcd_devision(a, b) {
    var tmp;

    while (b != 0) {
        tmp = b;
        b = a % b;
        a = tmp;
    }

    return a;
}

function gcd_subtraction(a, b) {
    while (a != b) {
        if (a > b) {
            a -= b;
        } else {
            b -= a;
        }
    }

    return a;
}

function gcd_recursive(a, b) {
    if (b == 0) {
        return a;
    } else {
        return gcd_recursive(b, a % b);
    }
}

File: Euclidean_algorithm/Go/jcla1/euclidean.go
package euclidean

func GCDDevision(a, b int) int {
	var tmp int

	for b != 0 {
		tmp = b
		b = a % b
		a = tmp
	}

	return a
}

func GCDSubtraction(a, b int) int {
	for a != b {
		if a > b {
			a -= b
		} else {
			b -= a
		}
	}

	return a
}

func GCDRecursive(a, b int) int {
	if b == 0 {
		return a
	} else {
		return gcd_recursive(b, a%b)
	}
}
